make.js

```diff
@@ -106,8 +106,11 @@ function makeApiFiles(api, sourceDir, apiOutputDir) {
         var testCppTemplate = getCompiledTemplate(path.resolve(sourceDir, "templates/Test.cpp.ejs"));
         writeFile(path.resolve(apiOutputDir, "test/TestApp/AutoGenTests/", "AutoGen" + api.name + "Tests.cpp"), testCppTemplate(locals));
 
+        var testLogCppTemplate = getCompiledTemplate(path.resolve(sourceDir, "templates/TestLog.cpp.ejs"));
+        writeFile(path.resolve(apiOutputDir, "test/TestApp/AutoGenTests/", "AutoGen" + api.name + "TestLog.cpp"), testLogCppTemplate(locals));
+
         var testDataCppTemplate = getCompiledTemplate(path.resolve(sourceDir, "templates/TestData.cpp.ejs"));
-        writeFile(path.resolve(apiOutputDir, "test/TestApp/AutoGenTests/", "AutoGen" + api.name + "TestData.cpp"), testDataCppTemplate(locals));
+        writeFile(path.resolve(apiOutputDir, "test/TestApp/AutoGenTests/", "AutoGen" + api.name + "TestData.cpp.autogen"), testDataCppTemplate(locals));

```

Comment:
>writeFile(path.resolve(apiOutputDir, "test/TestApp/AutoGenTests/", "AutoGen" + api.name + "TestData.cpp.autogen"), testDataCppTemplate(locals)); [](http://example.com/codeflow?start=8&length=144)

a little confused about exactly how this works.  It seems like SDKGenerator is outputting .cpp.autogen files.  Are those ever used? if not, should we skip outputting them to the SDK and just ejs compile them to somewhere in the template repo?

Reply:
they aren't used.  just put there so when some new API shows up we just grab it from that file.  otherwise we have to do more complex work to get to it.  i can dump them where ever, but having the template write the template repo isn't normal is it?

Reply:
maybe its better to one-time generate a testInput.json from the service swagger.  Then we could tweak that json file and generate the TestData.cpp files from that.  It seems like a bit of a weird in between to auto generate cpp files, and then manually change them.  If we ever need to regenerate them, your manual tweaks are all lost

Reply:
Chatted offline.  Going with this design.  Add .autogen files to gitignore.ejs for generated SDK